app/components/dashboard/run/left-panel/component.js

```diff
@@ -361,38 +340,44 @@ export default Component.extend(FullScreenMixin, {
           window.location.assign(url + '?token=' + token + '&taleFormat=' + format);
         },
         
-        openPublishModal(tale) {
-            const jwt = this.getDataONEJWT();
-            if (!jwt) {
-                // reroute to auth
-                $('#dataone-auth-modal').modal('show');
-                return;
-            }
-            
-            this.set('dataoneJWT', jwt);
+        authenticateD1(taleId, coordinatingNode) {
+          let callback = `${this.get('wholeTaleHost')}/run/${taleId}?auth=true`;
+          let endpoint = this.dataoneAuth.getPortalEndpoint(coordinatingNode)
+          endpoint += '/oauth?action=start&target=';
+          window.location.replace(endpoint + callback);
+      },
+
+        openPublishModal(tale) {          
             this.resetPublishState();
             this.set('taleToPublish', tale);
-            this.set('selectedRepository', null);
+            this.set('selectedRepositoryName', null);
             const self = this;
             $('#publish-modal').modal({ 
                 onApprove: () => false,
                 onDeny: () => false,
-                onVisible: () => { self.set('selectedRepository', self.get('repositories')[0].name); }
+                onVisible: () => { self.set('selectedRepositoryName', self.get('repositories')[0].name); }
             }).modal('show');
         },
         
-
         submitPublish(tale) {
             const self = this;
             console.log('Now publishing:', tale);
+
+            let targetRepo = self.get('selectedRepositoryName');
```

Comment:
Very minor observation, but couldn't you just use the selected index instead of getting the name and using a separate function to lookup by name?

Reply:
The semantic ui dropdown doesn't have a method for retrieving the index, but it does for text https://semantic-ui.com/modules/dropdown.html#/usage Scroll down to `Behavior`.
I use `get text` [here](https://github.com/whole-tale/dashboard/blob/ca4c565459d38c532ac19e2557d4569ce1c31a20/app/components/dashboard/run/left-panel/component.js#L405), every time the repository is changed. I made [getRepositoryFromName](https://github.com/whole-tale/dashboard/blob/ca4c565459d38c532ac19e2557d4569ce1c31a20/app/components/dashboard/run/left-panel/component.js#L219) as a way to get from the name, to the repository.


In 366 I could have used self.get('selectedRepository') and saved the call on 367 since it should be set by the time this code is reached.

Reply:
You can actually use the built-in prototype method `Array.find()` for this too.

`let selectedRepo = repositories.find((repo) => repo.name === selectedRepositoryName);` would do the same thing without needing to write/maintain a custom helper function for it.